tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuConverterConfig.java

```diff
@@ -0,0 +1,379 @@
+        }
+
+        /** Returns a converter config from the current builder state. */
+        public LdmlConverterConfig build() {
+            return new IcuConverterConfig(this);
+        }
+    }
+
+    private final Path cldrDir;
+    private final Path outputDir;
+    private final Path specialsDir;
+    private final ImmutableSet<OutputType> outputTypes;
+    private final CldrDraftStatus minimalDraftStatus;
+    private final boolean emitReport;
+
+    private IcuConverterConfig(Builder builder) {
+        this.cldrDir = checkNotNull(builder.cldrDir,
+            "must set a CLDR directory, or the CLDR_DIR system property");
+        if (DEFAULT_CLDR_DIR.isPresent() && !this.cldrDir.equals(DEFAULT_CLDR_DIR.get())) {
+            System.err.format(
+                "Warning: Specified CLDR base directory does not appear to match the"
+                    + " directory inferred by the 'CLDR_DIR' system property.\n"
+                    + "Specified: %s\n"
+                    + "Inferred: %s\n",
+                this.cldrDir, DEFAULT_CLDR_DIR.get());
+        }
+        this.outputDir = checkNotNull(builder.outputDir);
+        checkArgument(!Files.isRegularFile(outputDir),
+            "specified output directory if not a directory: %s", outputDir);
+        this.specialsDir = checkNotNull(builder.specialsDir,
+            "must specify a 'specials' XML directory");
+        checkArgument(Files.isDirectory(specialsDir),
+            "specified specials directory does not exist: %s", specialsDir);
+        this.outputTypes = builder.outputTypes;
+        checkArgument(!this.outputTypes.isEmpty(),
+            "must specify at least one output type to be generated (possible values are: %s)",
+            Arrays.asList(OutputType.values()));
+        this.minimalDraftStatus = builder.minimalDraftStatus;
+        this.emitReport = builder.emitReport;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override public Path getCldrDirectory() {
+        return cldrDir;
+    }
+
+    @Override public Path getOutputDir() {
+        return outputDir;
+    }
+
+    @Override public Set<OutputType> getOutputTypes() {
+        return outputTypes;
+    }
+
+    @Override public CldrDraftStatus getMinimumDraftStatus() {
+        return minimalDraftStatus;
+    }
+
+    @Override public Path getSpecialsDir() {
+        return specialsDir;
+    }
+
+    @Override public boolean emitReport() {
+        return emitReport;
+    }
+
+    // Currently hard-coded "hacks" which could be encoded via the builder if wanted.
+
+    @Override public Map<String, String> getForcedAliases(IcuLocaleDir dir) {
+        switch (dir) {
+        case COLL:
+            return ImmutableMap.<String, String>builder()
+                // It is not at all clear why this is being done (we expect "sr_Latn_ME" normally).
+                // TODO: Find out and document this properly.
+                .put("sr_ME", "sr_Cyrl_ME")
```

Comment:
is there a bug for this?

Reply:
I'm going to go through all by TODOs and generate about 20 tickets when this is all done (since during review some TODOs will be resolved, and I don't want to span a lot of tickets until after review).